dags/inference/maxtext_inference.py

```diff
@@ -16,204 +16,490 @@
+          "ici_parallelisms": [(1, 1, -1)],
+          "dataset": "openorca",
+          "num_prompts": 1000,
           "max_output_length": 1024,
+          "warmup_mode": "full",
       },
-      "llama2-13b": {
-          "sleep_time": 120,
-          "tpu_version_cores": [(TpuVersion.V5E, 8), (TpuVersion.V5P, 8)],
-          "checkpoint": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
-          "model_mode": "base",
-          "maxtext_logs": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/",
-          "scan_layers": "false",
-          "dataset": "openorca",
-          "weight_dtype": "bfloat16",
+      f"{LLAMA2_7B}-{W_BF16_KV_BF16}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0123-2013-2013",
+              "0213-0213-0213",
+              "0213-0213-0132",
+          ],
+      },
+      f"{LLAMA2_7B}-{W_INT8_KV_INT8}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0213-0213-0213",
+              "0213-0231-0213",
+          ],
+      },
+      # LLAMA2_13B
+      LLAMA2_13B: {
+          "maxtext_branch": maxtext_branch,
+          "jetstream_branch": jetstream_branch,
+          "sleep_time": 360,
+          "time_out_in_min": 120,
+          "tpu_version_cores": [(TpuVersion.V5E, 8)],#, (TpuVersion.V5E, 4)],
+          "model_name": LLAMA2_13B,
           "tokenizer": "tokenizer.llama2",
-          "per_device_batch_sizes": [1, 2, 4, 5, 6],
-          # (ici_fsdp_parallelism, ici_autoregressive_parallelism, ici_tensor_parallelism)
-          "ici_parallelisms": [(1, -1, 1), (1, 1, -1)],
-          "request_rate": 5,
-          "num_prompts": 1000,
+          "weight_dtype": "bfloat16",
+          "scan_layers": "false",
           "max_prefill_predict_length": 1024,
           "max_target_length": 2048,
+          "reshape_q": True,
+          # (ici_fsdp_parallelism, ici_autoregressive_parallelism, ici_tensor_parallelism)
+          "ici_parallelisms": [(1, 1, -1)],
+          "dataset": "openorca",
+          "request_rate": [0.0],
+          "num_prompts": 1000,
           "max_output_length": 1024,
+          "warmup_mode": "full",
       },
-      "llama2-70b": {
-          "sleep_time": 240,
+      f"{LLAMA2_13B}-{W_BF16_KV_BF16}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0123-1203-1203",  # baseline
+              "0213-0213-0213",  # default
+          ],
+      },
+      f"{LLAMA2_13B}-{W_INT8_KV_INT8}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0123-1203-1203",  # baseline
+              "0213-0213-0213",  # default
```

Comment:
@morgandu How about for the others below? I just copied the baseline and default for all the rest.

Reply:
Yep! This is good, the default layout can give good performance (may not be top 1) without layout tuning since it's on compute layout 0213, the baseline layout is what we have previously, and it's on compute layout 0123, where it requires a tuning.

Reply:
Thanks!